packages/next/build/entries.ts

```diff
@@ -29,6 +29,12 @@ export function createPagesMapping(
   hasServerComponents: boolean
 ): PagesMapping {
   const previousPages: PagesMapping = {}
+
+  // Do not process .d.ts files inside the `pages` folder
+  pagePaths = extensions.includes('ts')
+    ? pagePaths.filter((pagePath) => !pagePath.endsWith('.d.ts'))
```

Comment:
use `path.extension` instead of `endsWith`

Reply:
OK